RecoMET/METAlgorithms/src/HcalHaloAlgo.cc

```diff
@@ -119,6 +127,66 @@ HcalHaloData HcalHaloAlgo::Calculate(const CaloGeometry& TheCaloGeometry, edm::H
 	  TheHcalHaloData.GetPhiWedges().push_back( wedge );
 	}
     }
+
+  // Don't use HF.
+  int maxAbsIEta = 29;
+  std::vector<const CaloTower*> SortedCaloTowers;
+  for(CaloTowerCollection::const_iterator tower = TheCaloTowers->begin(); tower != TheCaloTowers->end(); tower++) {
+    if(abs(tower->ieta()) <= maxAbsIEta && tower->numProblematicHcalCells() > 0)
+      SortedCaloTowers.push_back(&(*tower));
+  }
+
+  // Sort towers such that lowest iphi and ieta are first, highest last, and towers
+  // with same iphi value are consecutive. Then we can do everything else in one loop.
+  std::sort(SortedCaloTowers.begin(), SortedCaloTowers.end(), CompareTowers);
+
+  std::vector<std::pair<char, CaloTowerDetId> > problematicStrip;
+  float stripHadEt = 0.0;
+
+  int prevIEta = -99, prevIPhi = -99;
+  float prevHadEt = 0.0;
+  std::pair<char, CaloTowerDetId> prevPair, towerPair;
+  bool wasContiguous = true;
+  // Loop through and store a vector of pairs (problematicCells, DetId) for each contiguous strip we find
+  for(unsigned int i = 0; i < SortedCaloTowers.size(); i++) {
+    const CaloTower* tower = SortedCaloTowers.at(i);
```

Comment:
``` sortedCaloTowers[i]; ```
.at() is too excessive considering you know already that i runs inside the range